auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -77,6 +90,14 @@ public void thisUsesUnstableApi() {}
   @Override
   public void applyRequestMetadata(MethodDescriptor<?, ?> method, Attributes attrs,
       Executor appExecutor, final MetadataApplier applier) {
+    SecurityLevel security = checkNotNull(attrs.get(ATTR_SECURITY_LEVEL), "securityLevel");
+    if (requirePrivacy && security != SecurityLevel.PRIVACY_AND_INTEGRITY) {
+      applier.fail(Status.UNAVAILABLE
```

Comment:
Should it be "FAILED_PRECONDITION"? This failure is not retry-able, while "UNAVAILABLE" implies retry-ability.

Reply:
"FAILED_PRECONDITION" does not imply non-retry-ability, and "UNAVAILABLE" does not always imply retry-ability, all depends on retry policy in service config. To disallow retry explicitly, send "grpc-retry-pushback-ms" trailer metadata with "-1".

Reply:
FAILED_PRECONDITION is currently black-listed for our use in https://github.com/grpc/grpc/blob/master/doc/statuscodes.md .

I don't disagree that UNAVAILABLE is not perfect, but I don't know what's better. If you're in this situation though, you're pretty hosed independent of the status code.